src/cljs/rems/util.cljs

```diff
@@ -171,3 +172,38 @@
          (fn []
            (.focus elem)
            false))))
+
+(defn- strip-trailing-zeroes
+  [s]
+  (let [without-decimal-zeroes (clojure.string/replace s #"\.[0]*$" "")
+        without-trailing-zeroes (clojure.string/replace s #"[0]+$" "")]
+    (if (and (= without-decimal-zeroes s)
+             (clojure.string/includes? s "."))
+      without-trailing-zeroes
+      without-decimal-zeroes)))
+
+(defn format-file-size
+  [size]
+  (when (or (zero? size) (pos? size))
+    (let [[file-size type] (condp > size
+                             (Math/pow 1000 2) [(/ size 1000.0) "KB"]
+                             (Math/pow 1000 3) [(/ size (Math/pow 1000 2)) "MB"]
+                             (Math/pow 1000 4) [(/ size (Math/pow 1000 3)) "GB"]
+                             [size "B"])]
+      (-> (format "%.2f" file-size)
+          strip-trailing-zeroes
+          (str " " type)))))
+
+(deftest test-format-file-size
+  (testing "should format sizes correctly"
+    (are [expected input] (= expected (format-file-size input))
+      "1 KB" 1000
+      "10 KB" (* 1000 10)
+      "1 GB" (* 1000 1000 1000)
+      "1.55 MB" (* 1000 1000 1.55)
+      "0.01 KB" (* 1000 0.012345)
+      "0 KB" 0
+      nil -1
+      nil nil
+      nil {}
+      nil [])))
```

Comment:
this is a little extra on top of the feature. i tried to cater to the test cases so that it's relatively obvious what is happening, but if it feels unnecessary or maintenance burden, we can drop it. in API we will implement the hard restriction and FE should basically give user some actionable feedback.

Reply:
Yeah, I can see what you mean by a little extra. It's ok to me. I have previously identified this library that could be used instead https://github.com/trhura/clojure-humanize not sure how great it actually would be.

Reply:
maybe if we could utilize other functions from the library it would be better than hand-made implementation. humanized output should be localizable though

Reply:
Yes, `humanized` is lacking in some areas like localization.